gwt-localizable-processor/src/main/java/org/gwtproject/i18n/processor/LocalizableProcessingStep.java

```diff
@@ -0,0 +1,151 @@
+package org.gwtproject.i18n.processor;
+
+import com.google.auto.common.BasicAnnotationProcessor;
+import com.google.auto.common.MoreElements;
+import com.google.common.collect.SetMultimap;
+import com.squareup.javapoet.*;
+import org.gwtproject.i18n.shared.Localizable;
+
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.util.Types;
+import javax.tools.Diagnostic;
+import java.io.IOException;
+import java.lang.annotation.Annotation;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class LocalizableProcessingStep implements BasicAnnotationProcessor.ProcessingStep {
+    private final ProcessingEnvironment processingEnv;
+
+    public LocalizableProcessingStep(ProcessingEnvironment processingEnv) {
+        this.processingEnv = processingEnv;
+    }
+
+    @Override
+    public Set<? extends Class<? extends Annotation>> annotations() {
+        return Collections.singleton(Localizable.IsLocalizable.class);
+    }
+
+    @Override
+    public Set<Element> process(SetMultimap<Class<? extends Annotation>, Element> elementsByAnnotation) {
+
+        for (Element element : elementsByAnnotation.get(Localizable.IsLocalizable.class)) {
+            // build a model
+            String packageName = processingEnv.getElementUtils().getPackageOf(element).getQualifiedName().toString();
+
+            // collect the possible locales
+            //TODO correctly sort this based on the logic ahmad has worked up
+
+            List<String> locales = getLocaleNames(element);
+            // TODO remove hack which restricts us to default
+            locales = Collections.singletonList("default");
+
+
+            // implement the interface into a new type for each locale key
+            Map<String, ClassName> keyToNameMapping = new HashMap<>();
+
+
+            keyToNameMapping.put("default", ClassName.get(packageName, element.getSimpleName().toString() + "_default"));
```

Comment:
Are we going to use the `_default` suffix or will go with default without suffix?!

Reply:
This line is just a hack to put only the `default` value in the map. In the future, the tooling would generate as many of the known locales as it could. 

That said, we require the `_default` suffix, since with no suffix it would have the same name as the original interface.